src/app/pages/TransactionDetailPage/index.tsx

```diff
@@ -32,39 +107,59 @@ export const TransactionDetailPage: FC = () => {
           <Skeleton variant="text" height={30} sx={{ my: 4 }} />
         </SubPageCard>
       )}
-      {transactions?.map(transaction => (
+      {transaction && (
         <SubPageCard title={t('transaction.header')} key={`${transaction.round}_${transaction.index}`}>
+          {!isMobile && <Divider variant="fullWidth" />}
           <StyledDescriptionList titleWidth="200px">
             <dt>{t('common.hash')}</dt>
-            <dd>{transaction.hash}</dd>
+            <dd>
+              {isMobile ? trimLongString(transaction.hash) : transaction.hash}
+              <CopyToClipboard value={transaction.hash} label={' '} />
+            </dd>
 
             <dt>{t('common.status')}</dt>
             <dd>
-              <TransactionStatusIcon success={transaction.success} />
+              <TransactionStatusIcon success={transaction.success} withText={true} />
             </dd>
 
             <dt>{t('common.block')}</dt>
-            <dd>{transaction.round}</dd>
+            <dd>
+              <Link component={RouterLink} to={RouteUtils.getBlockRoute(transaction.round, ParaTime.Emerald)}>
+                {transaction.round}
+              </Link>
+            </dd>
 
             <dt>{t('common.type')}</dt>
             <dd>
               <RuntimeTransactionLabel method={transaction.method} />
             </dd>
 
+            <dt>{t('common.timestamp')}</dt>
+            <dd>{formattedTimestamp}</dd>
+
             <dt>{t('common.from')}</dt>
-            <dd>{transaction.sender_0}</dd>
+            <dd>
+              <AccountLink address={transaction.sender_0} paratime={ParaTime.Emerald} />
+              <CopyToClipboard value={transaction.sender_0} label={' '} />
```

Comment:
At some point we should refactor `label={' '}`

Reply:
Yes, there are various possibilities here.
- What we want is displaying some potentially shortened text, potentially as a link, and a copy to clipboard icon.
- CopyToClipboard is able to display the value itself, but it doesn't support shortening the the text, or using a link.
- Supporting using shortened text is easy, but links are more complicated, because of the separation of concerns we are using between the MUI and react router. 
- So maybe it would be easiest to add the CopyToLink functionality to the (account) link widget itself....

Reply:
> At some point we should refactor label={' '}

Done, see https://github.com/oasisprotocol/explorer/pull/106